### DISCUSSION TITLE
Route to with an active route

### DISCUSSION THREAD

**Discussion by Esmolante** [organicmaps/organicmaps#2990] [Ideas]
I think it would be useful if, when we select "route to" with an active route, the old destination is converted in a stop instead of being deleted.
Or the option could be given to undo the existing route or not. For example, two buttons: "route to" and "continue to". 
In this way it would be easier to design complex routes (now it could only be done by setting the last stage from the beginning and adding stops).

**Comment by biodranik** [comment]
This option should not confuse users. We need to carefully think on the UX, and on the names/labels (button texts can't be too long)

**Comment by Esmolante** [comment]
Label of "continue to" could be the same of "route to".
Another option: if the "route to" is selected with an active route, a warning appears saying whether you want to delete the previous route or continue it to the new marked destination, with the previous destination becoming a stop. This saves a button. But I think a "Continue to" button is clearer.
Let me explain the situation in which this option might be necessary: you have created a complex route, with several stops. If, while you are designing it or driving it, another destination to go to comes up, now you would have to delete the route, choose the new final destination, and put all the intermediate stops again. A hassle. 
Another option, but more laborious to program (asking is easy when the work is done by others, sorry), would be an "edit route" option. And there you could add new stops and reorder or delete existing ones. And to top it off, but it is even more work, an option to optimize the route automatically reordering the stops (very useful for package distribution professionals).
